### PR TITLE
feat: add generic type to column payload

### PR DESCRIPTION
**Pull request by villebro**
### SUMMARY
Adds the generic datatype to the dataset column payload to avoid having to parse the raw datatypes in the frontend. Also bump `superset-ui` to bring in updates to `superset-ui/core` that are required to make this function properly: https://github.com/apache-superset/superset-ui/pull/1102

### BEFORE
Currently the numeric datatypes on the FCC 2018 example dataset show up as unidentified due to the frontend not being able to detect the generic datatype of the raw datatype from the database ("DOUBLE PRECISION"):
image

### AFTER
Now numeric types are being picked up automatically. In the future only datatypes that the db engine spec doesn't identify are shown as question marks:
image

### TEST PLAN
Added datatype tests for Postgres

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/datasource/ChangeDatasourceModal.tsx [hunk 1]
```diff
@@ -171,7 +171,7 @@ const ChangeDatasourceModal: FunctionComponent<ChangeDatasourceModalProps> = ({
 
   const handleChangeConfirm = () => {
     SupersetClient.get({
-      endpoint: `/datasource/get/${confirmedDataset?.type}/${confirmedDataset?.id}`,
+      endpoint: `/datasource/get/${confirmedDataset?.type}/${confirmedDataset?.id}/`,
```
Comment (villebro) [inline superset-frontend/src/datasource/ChangeDatasourceModal.tsx:174]: Bycatch - I noticed that the datasource request was always redirected to the one with the trailing slash.

**Comment by junlincc** [comment]
thank you Ville! for some reasons i am still seeing "?"  as the icon locally

**Comment by villebro** [comment]
> thank you Ville! for some reasons i am still seeing "?" as the icon locally

The question mark won't go away until https://github.com/apache-superset/superset-ui/pull/1102 is released and bumped on the main repo. But since this is not a breaking change I think it makes sense to merge this first to avoid having to make a `superset-ui` bump coincide with this PR.

**Comment by junlincc** [comment]
gotcha, @zhaoyongjie please help review, i wanna get this PR and apache-superset/superset-ui#1102 in soon. 🙏

File: superset/connectors/base/models.py [hunk 2]
```diff
@@ -533,6 +533,7 @@ class BaseColumn(AuditMixinNullable, ImportExportMixin):
     def __repr__(self) -> str:
         return str(self.column_name)
 
+    bool_types = ("BOOL",)
```
Comment (zhaoyongjie) [inline superset/connectors/base/models.py:536]: add more general column type
```
    bool_types = ("BOOL", "BOOLEAN", )
    num_types = ("TINYINT", )
    date_types = ("TIMESTAMP", )
    str_types = ("TEXT", "MEDIUMTEXT", "LONGTEXT", )
```

Comment (villebro) [reply superset/connectors/base/models.py:536]: Actually these aren't really used anywhere, it's just a fallback non-SQLA connectors; the models in `sqla/models.py` file checks these directly from the relevant db engine spec, hence the relevant definitions are defined in `BaseEngineSpec` and can be overridden on an engine-by-engine basis.

**Comment by junlincc** [comment]
hmmm still seeing "?"

File: superset-frontend/cypress-base/cypress/integration/dashboard/filter.test.ts [hunk 3]
```diff
@@ -30,7 +30,7 @@ describe('Dashboard filter', () => {
     cy.visit(WORLD_HEALTH_DASHBOARD);
   });
 
-  it('should apply filter', () => {
+  xit('should apply filter', () => {
```
Comment (villebro) [inline superset-frontend/cypress-base/cypress/integration/dashboard/filter.test.ts:33]: I've tested this manually, and can't reproduce the reason for this test to fail. Disabling for now as Filter Box is about to be deprecated soon (will take another stab at enabling this later once I get my local cypress env up and running).

Comment (villebro) [reply superset-frontend/cypress-base/cypress/integration/dashboard/filter.test.ts]: Turns out there was a timeout issue on cypress - fixed in the helper util

File: superset-frontend/cypress-base/cypress/integration/dashboard/dashboard.helper.ts [hunk 4]
```diff
@@ -45,7 +45,9 @@ export interface ChartSpec {
 
 export function getChartGridComponent({ name, viz }: ChartSpec) {
   return cy
-    .get(`[data-test="chart-grid-component"][data-test-chart-name="${name}"]`)
+    .get(`[data-test="chart-grid-component"][data-test-chart-name="${name}"]`, {
+      timeout: 30000,
+    })
```
Comment (villebro) [inline superset-frontend/cypress-base/cypress/integration/dashboard/dashboard.helper.ts:50]: This was just resolving very close to the default 5 secs, and apparently started causing widespread flakiness on cypress. I bumped this to the 30s seen a few lines below, now all tests pass.

**Review by zhaoyongjie — APPROVED** [review]
LGTM! thanks for the improvment.
Tested in postgres

image

File: superset-frontend/spec/javascripts/datasource/fixtures.tsx [hunk 5]
```diff
@@ -29,7 +29,8 @@ export const columns: ColumnMeta[] = [
     id: 516,
     is_dttm: false,
     python_date_format: null,
-    type: ColumnType.DOUBLE,
+    type: 'DOUBLE',
```
Comment (kgabryje) [inline superset-frontend/spec/javascripts/datasource/fixtures.tsx:32]: Why don't we use `ColumnType` here anymore? Comment applies to all other "ColumnType -> hardcoded string" changes

Comment (villebro) [reply superset-frontend/spec/javascripts/datasource/fixtures.tsx:32]: `ColumnType` has been removed from `superset-ui/core` and replaced with `string`, as it's the raw column type coming back from the database. `ColumnType` was really just a generalization that mostly complied with ANSI SQL types, but this was not a scalable solution, as the frontend mostly isn't aware of which database is being used. As an example, the datatype `DOUBLE PRECISION` wasn't being picked up on the frontend due to it being a datatype that's mostly only available on dialects of Postgres. In corner cases two datatypes with the same name can mean different things on different databases (the boolean type on MySQL comes to mind, which is actually an integer). As these are now handled in the backend and mapped to `GenericDataType`, that's what we're now going to be using for the most part.

**Pull request merged**